src/rollup/types.d.ts

```diff
@@ -783,9 +789,9 @@ export interface RollupWatchOptions extends InputOptions {
 	watch?: WatcherOptions | false;
 }
 
-interface TypedEventEmitter<T> {
+interface TypedEventEmitter<T extends {[event: string]: (...args: any) => any}> {
 	addListener<K extends keyof T>(event: K, listener: T[K]): this;
-	emit<K extends keyof T>(event: K, ...args: any[]): boolean;
+	emit<K extends keyof T>(event: K, ...args: Parameters<T[K]>): boolean;
```

Comment:
Ok, so I improved typechecking for events parameters and it shows some minor bugs, that's why in next lines I made input parameter optional.